source/operators/rekognition/check_content_moderation_status.py

```diff
@@ -115,9 +118,6 @@ def lambda_handler(event, context):
                             ContentModerationError="Unable to upload metadata for asset: {asset}".format(asset=asset_id))
                         output_object.add_workflow_metadata(PersonTrackingJobId=job_id)
                         raise MasExecutionError(output_object.return_output_object())
-                    output_object.add_workflow_metadata(ContentModerationJobId=job_id)
-                    output_object.update_workflow_status("Complete")
-                    return output_object.return_output_object()
```

Comment:
I did have some difficulty following the correctness of the termination condition (raise and returns) for all the branches in these rekognition functions, especially with the nested ifs.  I think we should consider refactoring, adding some comments, or even consolidating these functions so this isn't duplicated in so many places.  This comment is really for the existing code, rather than your change.